iocore/eventsystem/I_EventProcessor.h

```diff
@@ -307,6 +306,38 @@ class EventProcessor : public Processor
   volatile int n_dthreads       = 0; // No. of dedicated threads
   volatile int thread_data_used = 0;
   ink_mutex dedicated_spawn_thread_mutex;
+
+  /// Provide container style access to just the active threads, not the entire array.
+  struct active_threads_type {
+    using iterator = EThread **; ///< Internal iterator type, pointer to array element.
+    iterator _begin;             ///< Start of threads.
+    iterator _end;               ///< End of threads.
+    /// Construct from base of the array (@a start) and the current valid count (@a n).
+    active_threads_type(iterator start, int n) : _begin(start), _end(start + n) {}
+
+    iterator
+    begin() const
+    {
+      return _begin;
+    }
+    iterator
+    end() const
+    {
+      return _end;
+    }
+  };
+
+  // These can be used in container for loops and other range operations.
+  active_threads_type
+  active_ethreads()
```

Comment:
Subtle issue - if that is a `const` method then the iterator type must become `EThread * const *`.